src/address/AssetTransferAddress.ts

```diff
@@ -27,17 +45,37 @@ export class AssetTransferAddress {
             );
         }
 
-        if (type < 0x00 || type > 0x02) {
+        if (type < 0x00 || type > 0x03) {
             throw Error(`Unsupported type for asset transfer address: ${type}`);
         }
 
-        const words = toWords(
-            Buffer.from([
-                version,
-                type,
-                ...Buffer.from(H160.ensure(payload).value, "hex")
-            ])
-        );
+        let words;
+        if (H160.check(payload)) {
```

Comment:
How about extract these if-else statements to a function?

Reply:
I've changed it to call `function encodePayload(payload: PayloadValue): Buffer`.

```
const words = toWords(
            Buffer.from([version, type, ...encodePayload(payload)])
        );
```